Week04/src/BubbleSort.java

```diff
@@ -0,0 +1,17 @@
+public class BubbleSort {
+
+     public SalesRep[] BubbbleSort(SalesRep[] unsortedReps){
+
+         int n = unsortedReps.length;
+         for (int i = 0; i < n-1; i++){
```

Comment:
a do while would be better here instead of the first for, if you want to optimize with n-i-1 you can still keep the "i" as a plain int i; and not part of a for.